modules/context/context.go

```diff
@@ -249,6 +249,17 @@ func Contexter() macaron.Handler {
 		if ctx.Query("go-get") == "1" {
 			ownerName := c.Params(":username")
 			repoName := c.Params(":reponame")
+			if ownerName == "" || repoName == "" {
+				_, _ = c.Write([]byte(`<!doctype html>
```

Comment:
if returned error, you should give a 500 I think.

Reply:
Why not see the problem the other way and simply ignore go-get params instead of erroring ?
```
if ctx.Query("go-get") == "1" && ownerName != "" && repoName != "" {

}
```

Reply:
This could be break down in two `if` obviously.
```
if ctx.Query("go-get") == "1" 
  ownerName := c.Params(":username")
  repoName := c.Params(":reponame")
  if ownerName != "" && repoName != "" {
	... Do go-get work
  }
}
```

Reply:
But else we still need return 404 but not continue the other work.

Reply:
Once `ctx.Query("go-get") == "1" `, it should return anyway here.

Reply:
I think a 400 should be returned because in this case the client has not provided adequate information for the server to provide the client's request.

Reply:
I suppose https://github.com/go-gitea/gitea/blob/85202d4784758573f80f93dbbda97a444e7ece99/routers/repo/http.go#L58 should also be applied in this case

Reply:
Or perhaps in `trimmedRepoName`, such that this is not interfered with:
https://github.com/go-gitea/gitea/blob/bb609caceee95abfabfbb27bc2683339fe551fbf/modules/context/context.go#L258

Reply:
Why we should return an error ? For exemple github doesn't trigger an error on an profile if you use go-get param. Futhermore triggering an error from a supervision point could be misleading. I will not block has it need a quick fix.